### ISSUE TITLE
[WinUI] Style applied to `Label` displays different if using `Text` or `FormattedText`

### ISSUE THREAD

**Issue by AndreasReitberger** [dotnet/maui#13977]
### Description

I have  a `Label` control which uses a static style.
If setting the `Text` property, it looks like that.

```xaml
<Label
      HorizontalTextAlignment="Start"
      Margin="2,4"
      Style="{StaticResource LabelStyle}"
      VerticalTextAlignment="End"
      Text="{x:Static localization:Strings.Diskspace}"
      />
```

![image](https://user-images.githubusercontent.com/2952443/225555788-d00a64dd-a86e-4d8f-88bc-74efab05f6e7.png)

If setting the `FormattedText` instead, with the same style applied, it looks like that.

```xaml
<Label
      HorizontalTextAlignment="Start"
      Margin="2,4"
      Style="{StaticResource LabelStyle}"
      VerticalTextAlignment="End"
      >
      <Label.FormattedText>
          <FormattedString>
              <FormattedString.Spans>
                  <Span Text="{x:Static localization:Strings.Diskspace}" />
                  <!--<Span Text=" #1" />-->
              </FormattedString.Spans>
          </FormattedString>
      </Label.FormattedText>
  </Label>
```

![image](https://user-images.githubusercontent.com/2952443/225556707-f52b0693-d396-4993-99de-2ec68cb8d897.png)

Used style

```xaml
<Style x:Key="DefaultLabelStyle" TargetType="Label">
    <Setter Property="FontFamily" Value="{StaticResource MontserratSemiBold}" />
    <Setter Property="TextColor" Value="{AppThemeBinding Light={StaticResource Black}, Dark={StaticResource White}}" />
    <Setter Property="LineHeight" Value="{OnPlatform Default=1.25, iOS=-1}"/>
</Style>

<Style x:Key="LabelStyle" TargetType="Label" BasedOn="{StaticResource DefaultLabelStyle}">
    <Setter Property="FontSize" Value="{OnIdiom Desktop=14, Tablet=14, Default=12}"/>
</Style>
```

It seems that the `FontFamily` is not used correctly in this case

### Steps to Reproduce

1. Create a fresh MAUI app (from the template)
2. Add the label style and apply it once to a `Label` using `Text` and `FormattedText`
3. Use a custom font in the style (maybe a bold one?)
4. Observe the different in the style

### Link to public reproduction project repository

https://

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

Windows 10, net7.0-windows10.0.19041.0

**Comment by AndreasReitberger** [comment]
Hi @PureWeen 
I created a simple repro for this issue.

![image](https://user-images.githubusercontent.com/2952443/225818190-40fc43de-b40a-40e8-8c32-5f94382006c1.png)

Repro: https://github.com/AndreasReitberger/MauiIssue_13977

**Comment by bcaceiro** [comment]
Any news on this? It still happens in net8

**Comment by ppereira-serviceonsites** [comment]
also finding this happens

**Comment by pme442** [comment]
Any update on this?

**Comment by bcaceiro** [comment]
This is kinda important tbh

**Comment by RedZone908** [comment]
a workaround... you can apply the font family on the individual spans and it will work. It's yucky, but it's better than nothing.

**Comment by thisisthekap** [comment]
@RedZone908 While this issues needs fixing for sure, your workaround unblocked us in that aspect :) Thank you very much!

**Comment by dcsheive** [comment]
In Xamarin I used a data trigger to change fontfamily with a bool. It appears the span's fontfamily only reacts to its own fontfamily attribute. I bound the span's fontfamily to a string on the model and was able to change it that way.

**Comment by samhouts** [comment]
Yeah! 😥 We've been trying to prioritize issues with no workaround. We'll come back to this one!

**Comment by johan-broekhof** [comment]
I've ran into this issue aswell when migrating from Xamarin. I tried to solve this issue by creating a custom handler which inherits `LabelHandler` and added a method which maps the `FormattedText` property. I don't know whether this is a good solution, but filling the `FontFamily` property of each span when it is empty seems to do the trick:

```csharp
    private static void MapFormattedTextProperty(CustomLabelHandler handler, Label label)
    {
        var formattedText = label.FormattedText;
        var fontFamily = label.FontFamily;

        if (formattedText == null || fontFamily == null)
            return;

        foreach (var span in formattedText.Spans)
            span.FontFamily ??= fontFamily;
    }
```

**Comment by LeoJHarris** [comment]
This is quite an annoying issue that requires setting styles per `span` for it to work which IMO is not really a satisfactory work around as when/if this issue is resolved undoing that sort of change could take some effort depending on how many spans you are using in your application.

@iris-georgej thanks for providing your work around. Works for me.

**Comment by LeoJHarris** [comment]
@iris-georgej Updated your suggestion to include `TextColor` as this didn't seem to get applied in some situations.

```csharp
foreach (var span in label.FormattedText.Spans)
{
    span.Style ??= label.Style;
    span.TextColor ??= label.TextColor;
}
```
